docs/2020-06-22-Rewrite-Rule-Priorities.md

```diff
@@ -0,0 +1,236 @@
+Rewrite rules with priorities
+=============================
+
+Background
+----------
+
+Priorities allow writing case-based `K` code that closer resembles functional
+languages, i.e. it allows saying things like "Apply this rule first,
+if that doesn't work try this other one, if that also doesn't work try
+this, and so on". The `owise` attribute is a special case of this.
+
+### K syntax
+
+The `K` code looks roughly like this:
+
+```
+rule t(X) ⇒ s(X) requires P(X) [priority(<number>)]
+```
+
+where the highest priority is the one with the lowest number.
+
+As an example, if `A(X)` rewrites to
+* `B(X)` if `X<0`
+* `C(X)` if `X≥0 ∧ X<1`
+* `D(X)` if `X≥1`,
+we could write the following rules:
+
+```
+rule A(X) ⇒ B(X) requires X <Int 0 [priority(1)]
```

Comment:
why all the three rules rewrite to B(X)?

Reply:
Copy-paste error. Fixed.